src/Components/EventDetailsPage.js

```diff
@@ -82,34 +90,102 @@ class EventDetailsPage extends Component {
                 upvote_usernames: [],
                 upvote_userids: [],
                 comments_content: [],
+                comments_user: [],
                 comments_created: []
             };
             newPosts.push(newPost);
             this.setState({posts: newPosts, messageInput: ""});
         })
     }
 
+    toggleShowComments(threadId) {
+        let newShowComments = {};
+        newShowComments = Object.assign(newShowComments, this.state.showComments);
+        newShowComments[threadId] = !newShowComments[threadId];
+        this.setState({showComments: newShowComments});
+    }
+
+    handleSendComment(threadId, comment) {
+        const { eventId, token, user } = this.props.location;
+        const username = `${user.firstName} ${user.lastName}`;
+        axios.post(`http://localhost:8080/events/${eventId}/threads/${threadId}/comments`, {
+            token, content: comment, username
+        })
+        .then(res => {
+            if(res.status == 200) {
+                let newPosts = this.state.posts.slice();
+                let matchingThread = {};
+                let matchingThreadIndex = 0;
+                for(let i = 0; i < newPosts.length; i++) {
```

Comment:
Take a look at the lodash library which is a js utils library for deep copies. Might make this cleaner :) .
https://www.digitalocean.com/community/tutorials/js-deep-cloning-javascript-objects